hosts/pds3.py

```diff
@@ -0,0 +1,172 @@
+                        are appended to a list of standard cleaner functions.
+        no_standard:    If True, the standard cleaners are not used.
+    """
+
+    assert os.path.isfile(filespec), f'Not found: {filespec}'
+
+    # Find the label
+    labelspec = find_label(filespec)
+
+    # Load and clean the PDS label
+    lines = clean_label(pdsparser.PdsLabel.load_file(labelspec),
+                        cleaners=cleaners, no_standard=no_standard)
+
+    # Parse the label into a dictionary
+    return pdsparser.PdsLabel.from_string(lines).as_dict()
+
+
+################################################################################
+# UNIT TESTS
+################################################################################
+import unittest
+import os.path
+
+from hosts                   import pds3
+from oops.unittester_support import TESTDATA_PARENT_DIRECTORY
+
+
+#===========================================================================
+def clean_VIMS(line):
+    line = line.replace('(N/A',  '("N/A"')
+    line = line.replace('N/A)',  '"N/A")')
+    return line
+
+#===========================================================================
+def clean_UVIS(line):
+    if 'CORE_UNIT' in line:
+        if '"COUNTS/BIN"' not in line:
+            lines = line.replace('COUNTS/BIN', '"COUNTS/BIN"')
+    if line.startswith('ODC_ID'):
+        line = line.replace(',', '')
+    return line
+
+
+#===============================================================================
+class Test_PDS3(unittest.TestCase):
+
+
+    #===========================================================================
+    def runTest(self):
+
+        # Test extension replacement
+        filespec = 'cassini/ISS/W1575634136_1.IMG'
+        label_dict = pds3.get_label(os.path.join(TESTDATA_PARENT_DIRECTORY, filespec))
+        self.assertTrue(label_dict['PDS_VERSION_ID'] == 'PDS3')
+
+        # Test .LBL file input
+        filespec = 'cassini/ISS/W1575634136_1.LBL'
+        label_dict = pds3.get_label(os.path.join(TESTDATA_PARENT_DIRECTORY, filespec))
+        self.assertTrue(label_dict['PDS_VERSION_ID'] == 'PDS3')
+
+        # Test non-existent file
+        filespec = 'nofile.crap'
+        self.assertRaises(AssertionError, pds3.get_label, os.path.join(TESTDATA_PARENT_DIRECTORY, filespec))
+
+        # Test Cassini VIMS file
+        filespec = 'cassini/VIMS/v1793917030_1.lbl'
+        label_dict = pds3.get_label(os.path.join(TESTDATA_PARENT_DIRECTORY, filespec),
+                                    cleaners=[clean_VIMS])
+        self.assertTrue(label_dict['PDS_VERSION_ID'] == 'PDS3')
+
+        # Test Cassini UVIS file
+        filespec = 'cassini/UVIS/HSP2014_197_21_29.LBL'
+        label_dict = pds3.get_label(os.path.join(TESTDATA_PARENT_DIRECTORY, filespec),
+                                                 cleaners=[clean_UVIS])
+        self.assertTrue(label_dict['PDS_VERSION_ID'] == 'PDS3')
```

Comment:
Similar to what I have said above, the unit tests for each cleaner function belong inside that instrument module, not here.

Reply:
see above comment